plugin.rb

```diff
@@ -343,4 +343,23 @@ class Engine < ::Rails::Engine
 
   register_notification_consolidation_plan(reacted_by_two_users)
   register_notification_consolidation_plan(consolidated_reactions)
+
+  on(:post_moved) do |post, original_topic_id, original_post_id|
+    next if original_post_id.nil?
+    params = { old_post_id: original_post_id, new_post_id: post.id }
+
+    ActiveRecord::Base.transaction do
+      DB.exec(<<~SQL, params)
+        UPDATE discourse_reactions_reactions
+        SET post_id = :new_post_id
+        WHERE post_id = :old_post_id
+      SQL
+
+      DB.exec(<<~SQL, params)
+        UPDATE discourse_reactions_reaction_users
+        SET post_id = :new_post_id
+        WHERE post_id = :old_post_id
+      SQL
```

Comment:
I don't think this is right because the original post which still exists will lose its reactions in this case. In Discourse core when we move the first post, it actually copies all the existing `PostActions` so I think we need to follow that pattern as well.

https://github.com/discourse/discourse/blob/81c2d38cea7c805cca4dfc7f4245506554991f12/app/models/post_mover.rb#L261-L263